### DISCUSSION TITLE
Lesson 8 - How to withdraw ?

### DISCUSSION THREAD

**Discussion by getzelus** [Cyfrin/foundry-full-course-cu#661] [General]
Hello, 
On the HTML part at 2:50, I have an issue and a question. 

**I can't withdraw** with the first account of Anvil or an other (and i funded).

But it seems logic to me because only owner can do it. 
And the owner is the DeployFundMe contract no ? Or its automatically the user who deployed the first contract ? 

I tried to withdraw on Sepolia testnet and it works. 

I tried to cast call getOwner()(address) on my locally deployed contract and I get "contract does not exist" (even if anvil is running and its the right address) although the cast call works on Sepolia.

**Comment by shikhar229169** [comment]
First go through the below discussion to have better understanding of who the owner of FundMe is:
https://github.com/Cyfrin/foundry-full-course-f23/discussions/504

Now after reading that:
So, in case if you didn't put the vm.startBroadcast, then the owner will be the DeployFundMe contract.

for your last query about the cast doesn't work on anvil, it is because you didn't actually deployed your contract on anvil or you forgot to put --broadcast while deploying.

**Reply by getzelus** [reply]
Why do we need to add --broadcast, its not only to keep a trace in log files ? 
If we just write the command  forge deploy contractName, its not enough to deploy on local Anvil network (if its running) ? 

In fact I deployed (without --broadcast) a simple contract storage on anvil and could cast call a variable inside.

**Comment by olahamid** [comment]
hello @getzelus 
i'm currently in the same lesson and facing similar issue, for me the withdraw function doesn't withdraw the funds and also i was able to call it the withdraw function with any other address, hence the onlyOwner isn't is working. 

it's been a long time since you ask the question, do you sort out the problem or have an explanation to why it doesnt work.

**Reply by maxi7587** [reply]
Hi @olahamid ! The same happened to me, and code looked good. In my case, the issue was that I was not using the correct contact address.

When deploying FundMe with `make deploy`, if you look at the terminal logs, it creates 2 contracts:
```
##### anvil-hardhat
✅  [Success]Hash: 0x46fbc53f284599724c824c3930ed0c96ee44e9140d560002491f0b930975b9cd
Contract Address: 0xB7f8BC63BbcaD18155201308C8f3540b07f84F5e
Block: 12
Paid: 0.001462656183833988 ETH (451957 gas * 3.236272884 gwei)


##### anvil-hardhat
✅  [Success]Hash: 0xe12a43d22ffb37a5b9ee77a911620555877b3358da2c07613bb9a529c620c7a0
Contract Address: 0xA51c1fc2f0D1a1b8494Ed1FE312d7C3a78Ed91C0
Block: 13
Paid: 0.001849076027412762 ETH (576462 gas * 3.207628651 gwei)
```

But on top of that logs, right after **Script run successfully**, you have the contract name with the contract address that you must use in the Front End app:
```
Script ran successfully.

== Return ==
0: contract FundMe 0xA51c1fc2f0D1a1b8494Ed1FE312d7C3a78Ed91C0 <--- this is the correct address (you can see it matches one of the contacts)
```

When I used the address of the other contract (the first of the list), similar issues happened. But when using the correct address (from the 2nd contract), it works seamlessly.

Hope it helps!

**Reply by olahamid** [reply]
@maxi7587 thanks alot mate, I can't really remember if I ever sort out the error as it was a long time ago. I will have a look now and play around it. Thanks alot for letting me know the solution on it.
